.github/workflows/pipeline.yaml

```diff
@@ -0,0 +1,66 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - 'README.md'
+      - '.gitignore'
+      - 'documentation'
+  pull_request_review:
+    types:
+      - submitted
+  schedule:
+    - cron:  '30 17 * * *'
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout files 
+        uses: actions/checkout@v2
+      - name: Build docker
+        run: docker build --target test --platform linux/amd64 --tag todo-app:test .
+      - name: Run tests
+        run: docker run --env-file ./.env.test todo-app:test
+  publish:
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
+    name: Push to docker hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_ACCESS_USER }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Checkout files 
+        uses: actions/checkout@v2
+      - name: Build docker
+        run: docker build --target production --tag ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }} .
+      - name: Push to docker hub
+        run: docker push ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }}
+      - name: Heroku login credentials
+        run: |
+          cat > ~/.netrc <<EOF
+            machine api.heroku.com
+              login $HEROKU_EMAIL
+              password $HEROKU_API_KEY
+            machine git.heroku.com
+              login $HEROKU_EMAIL
+              password $HEROKU_API_KEY
+          EOF
+        env:
+          HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
+          HEROKU_EMAIL: ${{ secrets.HEROKU_EMAIL }}
+      - name: Add Heroku remote
+        run: heroku git:remote --app $HEROKU_APP_NAME
+        env:
+          HEROKU_APP_NAME: ${{ secrets.HEROKU_APP_NAME }}
+      - name: Login into Heroku's registry
+        run: heroku container:login
+      - name: Tag for heroku
+        run: docker tag ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }} registry.heroku.com/mh-todo-app/web
```

Comment:
You have HEROKU_APP_NAME configured as a secret above, but then hardcoded "mh-todo-app" here and in the push command